File: app/src/main/java/io/intrepid/skotlinton/screens/example2/Example2Presenter.java
package io.intrepid.skotlinton.screens.example2;

import android.support.annotation.NonNull;

import io.intrepid.skotlinton.base.BasePresenter;
import io.intrepid.skotlinton.base.PresenterConfiguration;
import io.intrepid.skotlinton.utils.RxUtils;
import io.reactivex.disposables.Disposable;

public class Example2Presenter extends BasePresenter<Example2Contract.View> implements Example2Contract.Presenter {

    Example2Presenter(@NonNull Example2Contract.View view,
                      @NonNull PresenterConfiguration configuration) {
        super(view, configuration);
    }

    @Override
    public void onViewCreated() {
        Disposable disposable = restApi.getMyIp()
                .compose(subscribeOnIoObserveOnUi())
                .subscribe(ipModel -> {
                    String ip = ipModel.ip;
                    view.showCurrentIpAddress(ip);
                    userSettings.setLastIp(ip);
                }, RxUtils.logError());
        disposables.add(disposable);

        String lastIp = userSettings.getLastIp();
        if (lastIp == null) {
            view.hidePreviousIpAddress();
        } else {
            view.showPreviousIpAddress(lastIp);
        }
    }
}

File: app/src/main/kotlin/io/intrepid/koroutines/screens/example2/Example2Presenter.kt
package io.intrepid.koroutines.screens.example2

import io.intrepid.koroutines.base.BasePresenter
import io.intrepid.koroutines.base.PresenterConfiguration
import io.intrepid.koroutines.utils.RxUtils
import io.reactivex.functions.Consumer

internal class Example2Presenter(view: Example2Contract.View, configuration: PresenterConfiguration)
    : BasePresenter<Example2Contract.View>(view, configuration), Example2Contract.Presenter {

    override fun onViewCreated() {
        super.onViewCreated()

        val disposable = restApi.getMyIp()
                .compose(subscribeOnIoObserveOnUi())
                .subscribe(Consumer {
                    val ip = it.ip
                    view?.showCurrentIpAddress(ip)
                    userSettings.lastIp = ip
                }, RxUtils.logError())
        disposables.add(disposable)

        val lastIp = userSettings.lastIp
        if (lastIp.isEmpty()) {
            view?.hidePreviousIpAddress()
        } else {
            view?.showPreviousIpAddress(lastIp)
        }
    }
}
